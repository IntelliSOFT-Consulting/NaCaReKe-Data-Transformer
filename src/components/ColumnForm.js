/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import PropTypes from 'prop-types';
import {
  Form, Button, Space, Select,
} from 'antd';
import { MinusCircleOutlined, PlusOutlined } from '@ant-design/icons';
import positions from '../data/positions';

const { Option } = Select;

export default function ColumnForm({
  cols,
  onFinish,
  handleChange,
  formRef,
  form,
}) {
  const opts = [
    ...new Set([...positions.rights, ...positions.yes, ...positions.lefts]),
  ];
  return (
    <Form
      name="dynamic_form_nest_item"
      onFinish={onFinish}
      autoComplete="off"
      ref={formRef}
      form={form}
      className="col-form"
    >
      <Form.List name="params">
        {(fields, { add, remove }) => (
          <>
            {fields.map(({
              key, name, fieldKey, ...restField
            }) => (
              <Space
                key={key}
                style={{ display: 'flex', marginBottom: 8 }}
                align="baseline"
              >
                <Form.Item
                  {...restField}
                  name={[name, 'check']}
                  fieldKey={[fieldKey, 'check']}
                  rules={[
                    {
                      required: true,
                      message: 'Missing column to be checked',
                    },
                  ]}
                >
                  <Select
                    showSearch
                    style={{ width: 200 }}
                    placeholder="Select column to check"
                    optionFilterProp="children"
                    filterOption={(input, option) => option.children
                      .toLowerCase()
                      .indexOf(input.toLowerCase()) >= 0}
                  >
                    {cols
                      && cols.map((col, i) => (
                        <Option value={col} key={i}>
                          {col}
                        </Option>
                      ))}
                  </Select>
                </Form.Item>

                <Form.Item
                  {...restField}
                  name={[name, 'update']}
                  fieldKey={[fieldKey, 'update']}
                  rules={[
                    {
                      required: true,
                      message: 'Missing column to be created',
                    },
                  ]}
                >
                  <Select
                    showSearch
                    style={{ width: 250 }}
                    mode="tags"
                    placeholder="Column to be created"
                    onChange={handleChange}
                  >
                    {opts.map((item) => (
                      <Option key={item}>{item}</Option>
                    ))}
                  </Select>
                </Form.Item>

                <MinusCircleOutlined onClick={() => remove(name)} />
              </Space>
            ))}
            <Form.Item>
              <Button
                type="dashed"
                onClick={() => add()}
                block
                icon={<PlusOutlined />}
                id="add-button"
              >
                Add column
              </Button>
            </Form.Item>
          </>
        )}
      </Form.List>
    </Form>
  );
}

ColumnForm.propTypes = {
  cols: PropTypes.arrayOf(PropTypes.string).isRequired,
  onFinish: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  formRef: PropTypes.object.isRequired,
  form: PropTypes.shape({}).isRequired,
};
