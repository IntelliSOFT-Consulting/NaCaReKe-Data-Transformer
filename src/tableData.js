/* eslint-disable import/no-mutable-exports */
let data = [
  [
    'RECS',
    'RECS (desc)',
    'CHEC',
    'CHEC (desc)',
    'AGE',
    'ADDR',
    'ADDR (cat)',
    'ADDR (desc)',
    'INCID',
    'TOP',
    'TOP (cat)',
    'TOP (desc)',
    'LTERALITY',
    'LTERALITY (desc)',
    'MOR',
    'MOR (desc)',
    'BEH',
    'BEH (desc)',
    'BAS',
    'BAS (desc)',
    'MPCODE',
    'MPSEQ',
    'MPTOT',
    'I10',
    'UPDATE',
    'ICCC',
    'OBSOLETEFLAGTUMOURTABLE',
    'TUMOURID',
    'PATIENTIDTUMOURTABLE',
    'PATIENTRECORDIDTUMOURTABLE',
    'TUMOURUPDATEDBY',
    'TUMOURUNDUPLICATIONSTATUS',
    'LABNUMBER',
    'LABREPORTAVAILABLEPTFILE',
    'LABREPORTAVAILABLEPTFILE (desc)',
    'CD4COUNT',
    'STATUSINDICATEDONCLINICALNOTES',
    'STATUSINDICATEDONCLINICALNOTES (desc)',
    'SPECIFICALLYNEGATIVE',
    'SPECIFICALLYNEGATIVE (desc)',
    'SPECIFICALLYPOSITIVE',
    'SPECIFICALLYPOSITIVE (desc)',
    'OTHERCONCURRENTILLNESS',
    'OTHERADDRESS',
    'GRADE',
    'GRADE (desc)',
    'STAGE',
    'STAGE (desc)',
    'REGNO',
    'PERS',
    'PERS (desc)',
    'FIRSTN',
    'MIDDLNAME',
    'FAMN',
    'SEX',
    'SEX (desc)',
    'BIRTHD',
    'TRIB',
    'TRIB (desc)',
    'OCCU',
    'OCCU (desc)',
    'DLC',
    'STAT',
    'STAT (desc)',
    'MIDN',
    'OBSOLETEFLAGPATIENTTABLE',
    'PATIENTRECORDID',
    'PATIENTUPDATEDBY',
    'PATIENTUPDATEDATE',
    'PATIENTRECORDSTATUS',
    'PATIENTCHECKSTATUS',
    'SURTCT',
    'SURTCT (desc)',
    'SURGERYDATE',
    'CHEMOTREAT',
    'CHEMOTREAT (desc)',
    'CHEMOTREATMENTDATE',
    'IMMUNOTREAT',
    'IMMUNOTREAT (desc)',
    'IMMUNOTREATMENTDATE',
    'HORMONETREATMENT',
    'HORMONETREATMENT (desc)',
    'HORMONETHERAPY',
    'RADIOTREAT',
    'RADIOTREAT (desc)',
    'RADIOTREATMENTDATE',
    'OTHERTREATMENT',
    'OTHERDATE',
    'HOSPITALNUMBER',
    'DEATHCAUSE',
    'REMARKSIFANY',
    'IDENTIFICATIONNO',
    'MARITALSTATUS',
    'MARITALSTATUS (desc)',
    'TELEPHONENO',
    'TELEPHONENUMBERNOK',
    'RELIGION',
    'RELIGION (desc)',
    'EDUCATIONLEVEL',
    'EDUCATIONLEVEL (desc)',
    'SMOKINGSTATUS',
    'SMOKINGSTATUS (desc)',
    'DRINKINGSTATUS',
    'DRINKINGSTATUS (desc)',
  ],
  [
    1,
    'Confirmed',
    1,
    'Done: OK',
    1,
    101,
    'Nairobi',
    'Westlands',
    2018010,
    250,
    'PANCREAS',
    'Head of pancreas',
    9,
    'Unknown',
    8140,
    'Adenocarcinoma, NOS',
    3,
    'Malignant',
    7,
    'Histology of primary',
    null,
    0,
    1,
    'C250',
    '25/11/2019',
    '11f',
    0,
    'KH0150890101',
    'KH015089',
    'KB43743847',
    'carolyne',
    null,
    null,
    9,
    'Unknown',
    null,
    9,
    'Unknown',
    9,
    'Unknown',
    9,
    'Unknown',
    null,
    null,
    9,
    'Unknown',
    9,
    'Unknown',
    'KH015089',
    null,
    'Invalid code.',
    'John',
    'Doe',
    'Jane',
    1,
    'Male',
    '99/99/2016',
    2,
    'Kangemi',
    8,
    'Child',
    '30/11/2018',
    1,
    'Alive',
    'John',
    null,
    'KB43743847',
    'Jean',
    '25/11/2018',
    0,
    null,
    9,
    'Unknown',
    null,
    2,
    'Yes',
    null,
    1,
    'Unknown',
    null,
    9,
    'Unknown',
    null,
    9,
    'Unknown',
    null,
    null,
    null,
    3293232,
    null,
    null,
    null,
    5,
    'Child',
    '7843434343',
    '73223928382 Michael',
    1,
    'Muslim',
    7,
    'Child',
    9,
    'Unknown',
    9,
    'Unknown',
  ],
];

const positions = {
  lefts: [
    'CHEMOTREAT (true)',
    'IMMUNOTREAT (true)',
    'HORMONETREATMENT (true)',
    'RADIOTREAT (true)',
    'SURGERY (true)',
    'HIV(true)',
    'SPECIFICALLYPOSITIVE (true)',
    'OTHERTREATMENT(TRUE)',
    'BAS (true)',
    'TELEPHONENO(+254)',
    'OTHERCONCURRENTILLNESS(true)',
  ],
  rights: [
    'MOR(Matching NCI Codes)',
    'TOP(Matching NCI Codes)',
    'ADDR (Sub County)',
    'WARD',
    'NOKNUMBER',
    'NOKNAME',
    'ADDR (County)',
  ],
  yes: [
    'CHEMOTREAT (true)',
    'IMMUNOTREAT (true)',
    'HORMONETREATMENT (true)',
    'RADIOTREAT (true)',
    'SURGERY (true)',
  ],
};

let visible = true;

const setVisible = () => {
  visible = !visible;
};

const setData = (datas) => {
  data = datas;
};

export { setData, data, visible, setVisible, positions };
