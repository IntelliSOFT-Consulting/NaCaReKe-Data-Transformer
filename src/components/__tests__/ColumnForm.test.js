/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import { data } from '../../tableData';
import ColumnForm from '../ColumnForm';

it('renders correctly', () => {
  const spy = sinon.spy();
  const wrapper = shallow(
    <ColumnForm
      handleChange={spy}
      cols={data[0]}
      onFinish={spy}
      formRef={{}}
      form={{}}
    />,
  );
  expect(wrapper).toMatchSnapshot();
});
