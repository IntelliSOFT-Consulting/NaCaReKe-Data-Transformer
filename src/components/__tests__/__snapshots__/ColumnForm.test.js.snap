// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ForwardRef(InternalForm)
  autoComplete="off"
  className="col-form"
  form={Object {}}
  name="dynamic_form_nest_item"
  onFinish={[Function]}
>
  <FormList
    name="params"
  >
    <Component />
  </FormList>
</ForwardRef(InternalForm)>
`;
